### PR TITLE
Week3

### PR DESCRIPTION
**Pull request by durgamanidevarapu**
please do code review

### CODE REVIEW

File: .mvn/wrapper/MavenWrapperDownloader.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
+/*
```
Comment (ashoksuryap) [inline .mvn/wrapper/MavenWrapperDownloader.java]: Remove maven generated file

File: src/main/java/com/product/review/reviewservice/ReviewServiceApplication.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.cloud.netflix.eureka.EnableEurekaClient;
+
+
+@SpringBootApplication
+@EntityScan
```
Comment (ashoksuryap) [inline src/main/java/com/product/review/reviewservice/ReviewServiceApplication.java]: Do we need this annotation?

File: src/main/java/com/product/review/reviewservice/ReviewServiceController.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        log.info("updated review detials:{}",review.toString());
+        return ResponseEntity.noContent().build();
+    }
+
+    private Review updateReviewEntity(ReviewDto reviewDto,Review review){
+
+        Review reviewEntity = convertToEntity(reviewDto);
+        reviewEntity.setId(review.getId());
+        if(StringUtils.isEmpty(reviewEntity.getDescription())){
+            reviewEntity.setDescription(review.getDescription());
```
Comment (ashoksuryap) [inline src/main/java/com/product/review/reviewservice/ReviewServiceController.java]: You need to update all your DB entity properties from dto . But, you are doing opposite to that. Please check

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,6 @@
+server.port=8086
+spring.jpa.hibernate.ddl-auto=none
+spring.jpa.properties.hibernate.id.new_generator_mappings=false
```
Comment (ashoksuryap) [inline src/main/resources/application.properties]: Do we need this?

File: src/main/java/com/product/review/reviewservice/ReviewServiceController.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+import java.util.Optional;
+
+
+@RestController
+public class ReviewServiceController {
+
+    @Autowired
+    ReviewService reviewService;
+    Logger log = LoggerFactory.getLogger(ReviewServiceController.class);
+    ModelMapper modelMapper = new ModelMapper();
```
Comment (ashoksuryap) [inline src/main/java/com/product/review/reviewservice/ReviewServiceController.java:28]: Please above fields as private

File: src/main/java/com/product/review/reviewservice/dao/ReviewServiceDao.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+@Repository
+public interface  ReviewServiceDao extends CrudRepository<Review, Long> {
+
+    @Query("select r from Review r where r.productId=:productId")
```
Comment (ashoksuryap) [inline src/main/java/com/product/review/reviewservice/dao/ReviewServiceDao.java]: Do we need @query for this? findByProductId is not getting list of Reviews by productId

File: src/main/java/com/product/review/reviewservice/service/ReviewServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+    @Override
+    public Review addProdcutReview(Review review) {
+        return reviewServiceDao.save(review);
+    }
+
+    @Override
+    public Iterable<Review> findProductReviews(int product_id) {
+
+        return reviewServiceDao.findByProductId(product_id);
```
Comment (ashoksuryap) [inline src/main/java/com/product/review/reviewservice/service/ReviewServiceImpl.java]: product_id ->productId

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+		<dependency>
+			<groupId>io.springfox</groupId>
+			<artifactId>springfox-swagger-ui</artifactId>
+			<version>2.9.2</version>
+		</dependency>
+		<!-- https://mvnrepository.com/artifact/com.google.code.gson/gson -->
+		<dependency>
+			<groupId>com.google.code.gson</groupId>
+			<artifactId>gson</artifactId>
+			<version>2.8.5</version>
```
Comment (ashoksuryap) [inline pom.xml:59]: Are you using this dependency?

File: .mvn/wrapper/MavenWrapperDownloader.java (hunk 1 above)
Comment (durgamanidevarapu) [reply .mvn/wrapper/MavenWrapperDownloader.java]: removed

File: src/main/java/com/product/review/reviewservice/ReviewServiceApplication.java (hunk 2 above)
Comment (durgamanidevarapu) [reply src/main/java/com/product/review/reviewservice/ReviewServiceApplication.java]: no need.. i  removed and committed

File: src/main/java/com/product/review/reviewservice/dao/ReviewServiceDao.java (hunk 6 above)
Comment (durgamanidevarapu) [reply src/main/java/com/product/review/reviewservice/dao/ReviewServiceDao.java]: no need.  removed and committed

File: src/main/java/com/product/review/reviewservice/service/ReviewServiceImpl.java (hunk 7 above)
Comment (durgamanidevarapu) [reply src/main/java/com/product/review/reviewservice/service/ReviewServiceImpl.java]: changed

File: pom.xml (hunk 8 above)
Comment (durgamanidevarapu) [reply pom.xml:59]: using in test class

File: src/main/resources/application.properties (hunk 4 above)
Comment (durgamanidevarapu) [reply src/main/resources/application.properties]: no.. removed

File: src/main/java/com/product/review/reviewservice/ReviewServiceController.java (hunk 5 above)
Comment (durgamanidevarapu) [reply src/main/java/com/product/review/reviewservice/ReviewServiceController.java:28]: modified

File: src/main/java/com/product/review/reviewservice/ReviewServiceController.java (hunk 3 above)
Comment (durgamanidevarapu) [reply src/main/java/com/product/review/reviewservice/ReviewServiceController.java]: changed the code

**Review by ashoksuryap — APPROVED** [review]

**Pull request merged**